File: backendApp/src/main/java/com/cinemas/entities/CinemaImages.java
package com.cinemas.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "cinema_images")
public class CinemaImages {
    @Id
    @Column(name = "uid")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer uid;

    @Column()
    private String url;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "cinema_id", referencedColumnName = "id")
    Cinema cinema;

}

File: fllmbooking_app/lib/data/models/CinemaImages.dart
class CinemaImages{
  int? uid;
  String? url;
  String? cinemaId;

  CinemaImages({required this.uid, required this.url, required this.cinemaId});

  factory CinemaImages.fromJson(Map<String, dynamic> json) {
    return CinemaImages(
      uid: json['uid'],
      url: json['url'],
      cinemaId: json['cinemaId'],
    );
  }
}